server/exports.lua

```diff
@@ -317,19 +317,25 @@ end
 QBCore.Functions.GetCoreVersion = GetCoreVersion
 exports('GetCoreVersion', GetCoreVersion)
 
-local function ExploitBan(playerId, origin)
-    local name = GetPlayerName(playerId)
+-- Player Ban
+local function BanPlayer(src,reason,expire,resource)
+    local name = GetPlayerName(src)
     MySQL.insert('INSERT INTO bans (name, license, discord, ip, reason, expire, bannedby) VALUES (?, ?, ?, ?, ?, ?, ?)', {
         name,
-        QBCore.Functions.GetIdentifier(playerId, 'license'),
-        QBCore.Functions.GetIdentifier(playerId, 'discord'),
-        QBCore.Functions.GetIdentifier(playerId, 'ip'),
-        origin,
-        2147483647,
-        'Anti Cheat'
+        QBCore.Functions.GetIdentifier(src, 'license'),
+        QBCore.Functions.GetIdentifier(src, 'discord'),
+        QBCore.Functions.GetIdentifier(src, 'ip'),
+        reason,
+        expire,
+        resource
     })
-    DropPlayer(playerId, Lang:t('info.exploit_banned', {discord = QBCore.Config.Server.Discord}))
-    TriggerEvent("qb-log:server:CreateLog", "anticheat", "Anti-Cheat", "red", name .. " has been banned for exploiting " .. origin, true)
+    TriggerEvent('qb-log:server:CreateLog', resource, 'Player Banned', 'red', string.format('%s was banned by %s for %s', name, resource, reason), true)
+    DropPlayer(src, string.format("You were permanently banned by the server for: %s",reason))
 end
+exports("BanPlayer",BanPlayer)
 
-exports('ExploitBan', ExploitBan)
+-- Deprecated exploit ban
+local function ExploitBan(playerId, origin)
+    BanPlayer(playerId,origin,"2147483647",'Anti Cheat')
```

Comment:
What's the number as a string?

Reply:
I'm guessing it's an epoch to ban a user infefinately..
Just looks like a weird number is all

I realise you've only moved it here btw, just interested.

Reply:
That's exactly what it is. I could take it out of quotes.

Reply:
After some googling, it also seems like its the largest possible int32 number.

Reply:
It is the epoch date to release ban...